class/User.js

```diff
@@ -426,17 +433,42 @@ export class User {
     }
   }
 
+  async _getChainTransactions() {
+    return new Promise((resolve, reject) => {
+      this._lightning.getTransactions({}, (err, data) => {
+        if (err) return reject(err);
+        const { transactions } = data;
+        const outTxns = [];
+        // on lightning incoming transactions have no labels
+        // for now filter out known labels to reduce transactions
+        transactions
+          .filter((tx) => tx.label !== 'external' && !tx.label.includes('openchannel'))
+          .map((tx) => {
+            const decodedTx = decodeRawHex(tx.raw_tx_hex);
+            decodedTx.outputs.forEach((vout) =>
+              outTxns.push({
+                // mark all as received, since external is filtered out
+                category: 'receive',
+                confirmations: tx.num_confirmations,
+                amount: Number(vout.value),
+                address: vout.scriptPubKey.addresses[0],
+                time: tx.time_stamp,
+              }),
+            );
+          });
+
+        resolve(outTxns);
+      });
+    });
+  }
+
   /**
    * Returning onchain txs for user's address that are less than 3 confs
    *
    * @returns {Promise<Array>}
    */
   async getPendingTxs() {
-    let addr = await this.getAddress();
-    if (!addr) {
-      await this.generateAddress();
-      addr = await this.getAddress();
-    }
+    const addr = await this.getOrGenerateAddress();
     if (!addr) throw new Error('cannot get transactions: no onchain address assigned to user');
```

Comment:
this string is now inside  `getOrGenerateAddress()` so it should be safe to remove it from here